Sources/EditController.swift

```diff
@@ -319,9 +319,23 @@ extension EditController: AddToPlaylistViewControllerDelegate {
     func addToPlaylistViewController(_ addToPlaylistViewController: AddToPlaylistViewController,
                                      didSelectPlaylist playlist: VLCMLPlaylist) {
         let files = model.anyfiles
+
         for index in selectedCellIndexPaths where index.row < files.count {
-            if !playlist.appendMedia(withIdentifier: files[index.row].identifier()) {
-                assertionFailure("EditController: AddToPlaylistViewControllerDelegate: Failed to add item.")
+            if let mediaCollection = files[index.row] as? MediaCollectionModel {
+                guard let collectionFiles = mediaCollection.files() else {
+                    // unable to retreive files from a media collection
+                    continue
+                }
```

Comment:
I don't think you need the guard here, you can just say ```for file in mediaCollection.files()``` if it's nil it will just not jump into the for loop, which has the same effect as the continue but I guess the shortest is something like this
```
let mediaCollection = files[index.row] as? MediaCollectionModel
let filesToAppend = mediaCollection != nil ? mediaCollection.files() : [files[index.row]]
for file in filesToAppend {
    if !playlist.appendMedia(withIdentifier: file.identifier()) {
        assertionFailure("EditController: AddToPlaylistViewControllerDelegate: Failed to add item.")
    }
}